offload/DeviceRTL/src/Utils.cpp

```diff
@@ -57,6 +59,12 @@ int32_t shuffleDown(uint64_t Mask, int32_t Var, uint32_t LaneDelta,
   return __builtin_amdgcn_ds_bpermute(Index << 2, Var);
 }
 
+uint64_t ballotSync(uint64_t Mask, int32_t Pred) {
+  return Mask &
+         (__AMDGCN_WAVEFRONT_SIZE == 32 ? __builtin_amdgcn_ballot_w32(Pred)
+                                        : __builtin_amdgcn_ballot_w64(Pred));
```

Comment:
```suggestion
  return Mask & __builtin_amdgcn_ballot_w64(Pred));
```
The `w64` intrinsic works on both targets, the backend lowers it correctly. See https://godbolt.org/z/Ph6sn3Mh6.

Reply:
Oh interesting. :-) Good to know.